{"name": "test_sauce_demo_invalid_login", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n0   chromedriver                        0x00000001029df184 cxxbridge1$str$ptr + 3626716\n1   chromedriver                        0x00000001029d79d4 cxxbridge1$str$ptr + 3596076\n2   chromedriver                        0x0000000102444968 cxxbridge1$string$len + 89228\n3   chromedriver                        0x0000000102488d50 cxxbridge1$string$len + 368756\n4   chromedriver                        0x00000001024c25b4 cxxbridge1$string$len + 604376\n5   chromedriver                        0x000000010247d568 cxxbridge1$string$len + 321676\n6   chromedriver                        0x000000010247e1b8 cxxbridge1$string$len + 324828\n7   chromedriver                        0x00000001029aa9ac cxxbridge1$str$ptr + 3411716\n8   chromedriver                        0x00000001029adccc cxxbridge1$str$ptr + 3424804\n9   chromedriver                        0x000000010299186c cxxbridge1$str$ptr + 3308996\n10  chromedriver                        0x00000001029ae58c cxxbridge1$str$ptr + 3427044\n11  chromedriver                        0x000000010298309c cxxbridge1$str$ptr + 3249652\n12  chromedriver                        0x00000001029c84b8 cxxbridge1$str$ptr + 3533328\n13  chromedriver                        0x00000001029c8634 cxxbridge1$str$ptr + 3533708\n14  chromedriver                        0x00000001029d7648 cxxbridge1$str$ptr + 3595168\n15  libsystem_pthread.dylib             0x00000001888232e4 _pthread_start + 136\n16  libsystem_pthread.dylib             0x000000018881e0fc thread_start + 8", "trace": "driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"ee4e1e1f1b7b37a6348397473f733481\")>\n\n    @allure.feature(\"Invalid Login Handling\")  # Group related test cases\n    @allure.story(\"Incorrect Password\")  # D\n    def test_sauce_demo_invalid_login(driver):\n        \"\"\"Test case for the invalid login flow in SauceDemo.\"\"\"\n        automation = SauceDemoAutomation(driver)\n    \n        with allure.step(\"Open SauceDemo website\"):\n            automation.open_website(\"https://www.saucedemo.com/\")\n    \n        with allure.step(\"Log in to SauceDemo\"):\n            automation.login(SAUCEDO_USERNAME, \"invalid_password\")\n    \n        with allure.step(\"Validate error message\"):\n>           automation.validate_error_msg()\n\ntests/main.py:71: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \ntests/automation.py:76: in validate_error_msg\n    error_msg =self.wait.until(EC.presence_of_element_located((By.XPATH, \"//button[@data-test=error-button]\")))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"ee4e1e1f1b7b37a6348397473f733481\")>\nmethod = <function presence_of_element_located.<locals>._predicate at 0x1031b3740>\nmessage = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            if time.monotonic() > end_time:\n                break\n            time.sleep(self._poll)\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       0   chromedriver                        0x00000001029df184 cxxbridge1$str$ptr + 3626716\nE       1   chromedriver                        0x00000001029d79d4 cxxbridge1$str$ptr + 3596076\nE       2   chromedriver                        0x0000000102444968 cxxbridge1$string$len + 89228\nE       3   chromedriver                        0x0000000102488d50 cxxbridge1$string$len + 368756\nE       4   chromedriver                        0x00000001024c25b4 cxxbridge1$string$len + 604376\nE       5   chromedriver                        0x000000010247d568 cxxbridge1$string$len + 321676\nE       6   chromedriver                        0x000000010247e1b8 cxxbridge1$string$len + 324828\nE       7   chromedriver                        0x00000001029aa9ac cxxbridge1$str$ptr + 3411716\nE       8   chromedriver                        0x00000001029adccc cxxbridge1$str$ptr + 3424804\nE       9   chromedriver                        0x000000010299186c cxxbridge1$str$ptr + 3308996\nE       10  chromedriver                        0x00000001029ae58c cxxbridge1$str$ptr + 3427044\nE       11  chromedriver                        0x000000010298309c cxxbridge1$str$ptr + 3249652\nE       12  chromedriver                        0x00000001029c84b8 cxxbridge1$str$ptr + 3533328\nE       13  chromedriver                        0x00000001029c8634 cxxbridge1$str$ptr + 3533708\nE       14  chromedriver                        0x00000001029d7648 cxxbridge1$str$ptr + 3595168\nE       15  libsystem_pthread.dylib             0x00000001888232e4 _pthread_start + 136\nE       16  libsystem_pthread.dylib             0x000000018881e0fc thread_start + 8\n\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:105: TimeoutException"}, "description": "Test case for the invalid login flow in SauceDemo.", "steps": [{"name": "Open SauceDemo website", "status": "passed", "steps": [{"name": "Open website: 'https://www.saucedemo.com/'", "status": "passed", "parameters": [{"name": "url", "value": "'https://www.saucedemo.com/'"}], "start": 1735178387005, "stop": 1735178389216}], "start": 1735178387004, "stop": 1735178389216}, {"name": "Log in to SauceDemo", "status": "passed", "steps": [{"name": "Login with username: 'standard_user' and password: 'invalid_password'", "status": "passed", "parameters": [{"name": "username", "value": "'standard_user'"}, {"name": "password", "value": "'invalid_password'"}], "start": 1735178389216, "stop": 1735178389464}], "start": 1735178389216, "stop": 1735178389464}, {"name": "Validate error message", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n0   chromedriver                        0x00000001029df184 cxxbridge1$str$ptr + 3626716\n1   chromedriver                        0x00000001029d79d4 cxxbridge1$str$ptr + 3596076\n2   chromedriver                        0x0000000102444968 cxxbridge1$string$len + 89228\n3   chromedriver                        0x0000000102488d50 cxxbridge1$string$len + 368756\n4   chromedriver                        0x00000001024c25b4 cxxbridge1$string$len + 604376\n5   chromedriver                        0x000000010247d568 cxxbridge1$string$len + 321676\n6   chromedriver                        0x000000010247e1b8 cxxbridge1$string$len + 324828\n7   chromedriver                        0x00000001029aa9ac cxxbridge1$str$ptr + 3411716\n8   chromedriver                        0x00000001029adccc cxxbridge1$str$ptr + 3424804\n9   chromedriver                        0x000000010299186c cxxbridge1$str$ptr + 3308996\n10  chromedriver                        0x00000001029ae58c cxxbridge1$str$ptr + 3427044\n11  chromedriver                        0x000000010298309c cxxbridge1$str$ptr + 3249652\n12  chromedriver                        0x00000001029c84b8 cxxbridge1$str$ptr + 3533328\n13  chromedriver                        0x00000001029c8634 cxxbridge1$str$ptr + 3533708\n14  chromedriver                        0x00000001029d7648 cxxbridge1$str$ptr + 3595168\n15  libsystem_pthread.dylib             0x00000001888232e4 _pthread_start + 136\n16  libsystem_pthread.dylib             0x000000018881e0fc thread_start + 8\n\n", "trace": "  File \"/Users/eitanpeles/Documents/saucedemo/tests/main.py\", line 71, in test_sauce_demo_invalid_login\n    automation.validate_error_msg()\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/eitanpeles/Documents/saucedemo/tests/automation.py\", line 76, in validate_error_msg\n    error_msg =self.wait.until(EC.presence_of_element_located((By.XPATH, \"//button[@data-test=error-button]\")))\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/selenium/webdriver/support/wait.py\", line 105, in until\n    raise TimeoutException(message, screen, stacktrace)\n"}, "steps": [{"name": "Validate error message", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n0   chromedriver                        0x00000001029df184 cxxbridge1$str$ptr + 3626716\n1   chromedriver                        0x00000001029d79d4 cxxbridge1$str$ptr + 3596076\n2   chromedriver                        0x0000000102444968 cxxbridge1$string$len + 89228\n3   chromedriver                        0x0000000102488d50 cxxbridge1$string$len + 368756\n4   chromedriver                        0x00000001024c25b4 cxxbridge1$string$len + 604376\n5   chromedriver                        0x000000010247d568 cxxbridge1$string$len + 321676\n6   chromedriver                        0x000000010247e1b8 cxxbridge1$string$len + 324828\n7   chromedriver                        0x00000001029aa9ac cxxbridge1$str$ptr + 3411716\n8   chromedriver                        0x00000001029adccc cxxbridge1$str$ptr + 3424804\n9   chromedriver                        0x000000010299186c cxxbridge1$str$ptr + 3308996\n10  chromedriver                        0x00000001029ae58c cxxbridge1$str$ptr + 3427044\n11  chromedriver                        0x000000010298309c cxxbridge1$str$ptr + 3249652\n12  chromedriver                        0x00000001029c84b8 cxxbridge1$str$ptr + 3533328\n13  chromedriver                        0x00000001029c8634 cxxbridge1$str$ptr + 3533708\n14  chromedriver                        0x00000001029d7648 cxxbridge1$str$ptr + 3595168\n15  libsystem_pthread.dylib             0x00000001888232e4 _pthread_start + 136\n16  libsystem_pthread.dylib             0x000000018881e0fc thread_start + 8\n\n", "trace": "  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/eitanpeles/Documents/saucedemo/tests/automation.py\", line 76, in validate_error_msg\n    error_msg =self.wait.until(EC.presence_of_element_located((By.XPATH, \"//button[@data-test=error-button]\")))\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/selenium/webdriver/support/wait.py\", line 105, in until\n    raise TimeoutException(message, screen, stacktrace)\n"}, "start": 1735178389464, "stop": 1735178399783}], "start": 1735178389464, "stop": 1735178399786}], "start": 1735178387004, "stop": 1735178399786, "uuid": "9f0d2d6a-88e6-4c8b-a16c-08d01faa5c9f", "historyId": "3818446b06c10f618db8fdd812617c07", "testCaseId": "3818446b06c10f618db8fdd812617c07", "fullName": "tests.main#test_sauce_demo_invalid_login", "labels": [{"name": "feature", "value": "Invalid Login Handling"}, {"name": "story", "value": "Incorrect Password"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "main"}, {"name": "host", "value": "ip-192-168-15-3.eu-central-1.compute.internal"}, {"name": "thread", "value": "54231-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.main"}]}